packages/babel-helpers/src/helpers/applyDecs2301.js

```diff
@@ -300,392 +274,437 @@ function applyMemberDec(
         assertValidReturnValue(kind, newValue);
-        var newInit;
 
         if (kind === 0 /* FIELD */) {
-          newInit = newValue;
+          init = newValue;
         } else if (kind === 1 /* ACCESSOR */) {
-          newInit = newValue.init;
+          init = newValue.init;
           get = newValue.get || value.get;
           set = newValue.set || value.set;
 
           value = { get: get, set: set };
         } else {
           value = newValue;
         }
+      }
+    } else {
+      for (var i = decs.length - 1; i >= 0; i--) {
+        var dec = decs[i];
+
+        newValue = memberDec(
+          dec,
+          name,
+          desc,
+          initializers,
+          kind,
+          isStatic,
+          isPrivate,
+          value,
+          hasPrivateBrand
+        );
 
-        if (newInit !== void 0) {
-          if (init === void 0) {
-            init = newInit;
-          } else if (typeof init === "function") {
-            init = [init, newInit];
+        if (newValue !== void 0) {
+          assertValidReturnValue(kind, newValue);
+          var newInit;
+
+          if (kind === 0 /* FIELD */) {
+            newInit = newValue;
+          } else if (kind === 1 /* ACCESSOR */) {
+            newInit = newValue.init;
+            get = newValue.get || value.get;
+            set = newValue.set || value.set;
+
+            value = { get: get, set: set };
           } else {
-            init.push(newInit);
+            value = newValue;
+          }
+
+          if (newInit !== void 0) {
+            if (init === void 0) {
+              init = newInit;
+            } else if (typeof init === "function") {
+              init = [init, newInit];
+            } else {
+              init.push(newInit);
+            }
           }
         }
       }
     }
-  }
-
-  if (kind === 0 /* FIELD */ || kind === 1 /* ACCESSOR */) {
-    if (init === void 0) {
-      // If the initializer was void 0, sub in a dummy initializer
-      init = function (instance, init) {
-        return init;
-      };
-    } else if (typeof init !== "function") {
-      var ownInitializers = init;
 
-      init = function (instance, init) {
-        var value = init;
+    if (kind === 0 /* FIELD */ || kind === 1 /* ACCESSOR */) {
+      if (init === void 0) {
+        // If the initializer was void 0, sub in a dummy initializer
+        init = function (instance, init) {
+          return init;
+        };
+      } else if (typeof init !== "function") {
+        var ownInitializers = init;
 
-        for (var i = 0; i < ownInitializers.length; i++) {
-          value = ownInitializers[i].call(instance, value);
-        }
+        init = function (instance, init) {
+          var value = init;
 
-        return value;
-      };
-    } else {
-      var originalInitializer = init;
+          for (var i = 0; i < ownInitializers.length; i++) {
+            value = ownInitializers[i].call(instance, value);
+          }
```

Comment:
Can we use `for of` in helpers? I'm not sure if it's allowed in helpers.

Reply:
We try to keep helpers es5-only _when possible_ (or es5 syntax + polyfillable functions)